src/codegen/generators/statement_generator.rs

```diff
@@ -1,12 +1,18 @@
+use std::ops::Range;
+
 use super::{expression_generator::ExpressionCodeGenerator, llvm::LLVM};
 use crate::codegen::llvm_typesystem::cast_if_needed;
 use crate::codegen::LLVMTypedIndex;
-/// Copyright (c) 2020 Ghaith Hachem and Mathias Rieder
-use crate::index::{ImplementationIndexEntry, Index};
+use crate::typesystem::{RANGE_CHECK_LS_FN, RANGE_CHECK_LU_FN, RANGE_CHECK_S_FN, RANGE_CHECK_U_FN};
 use crate::{
     ast::{flatten_expression_list, ConditionalBlock, Operator, Statement},
     compile_error::CompileError,
 };
+/// Copyright (c) 2020 Ghaith Hachem and Mathias Rieder
```

Comment:
Move this back up. 
It this keeps moving it might be the wrong formatting for that line, we should investigate that at some point.